### DISCUSSION TITLE
How do I use BSgenome.Hsapiens.NCBI.GRCh38

### DISCUSSION THREAD

**Discussion by wshands** [GreenleafLab/ArchR#1018] [Questions / Documentation] [answered]
I would like to use BSgenome.Hsapiens.NCBI.GRCh38 with ArchR
I don't have a minimal background in bioinformatics and am working through the documentation at https://www.archrproject.com/bookdown/getting-set-up.html

I have done the following:
BiocManager::install("BSgenome.Hsapiens.NCBI.GRCh38")
library("BSgenome.Hsapiens.NCBI.GRCh38")
genomeAnnotation <- createGenomeAnnotation(genome = BSgenome.Hsapiens.NCBI.GRCh38)

The documentation then says that I need
_A TxDb object (transcript database) from Bioconductor which contains information for gene/transcript coordinates.
An OrgDb object (organism database) from Bioconductor which provides a unified framework to map between gene names and various gene identifiers._

However I don't find TxDb.Hsapiens.NCBI.GRCh38.knownGene on BioConductor (I do find org.Hs.eg.db on BioConductor). How do I get around this? Must I follow these instructions?

_Alternatively, if you dont have a TxDb and OrgDb object, you can create a geneAnnotation object from the following information :_

_A “genes” object - a GRanges object containing gene coordinates (start to end). This object must have a symbols column matching the symbols column of the “exons” object described below.
An “exons” object - GRanges object containing gene exon coordinates. Must have a symbols column matching the symbols column of the “genes” object described above.
A GRanges object containing standed transcription start site (TSS) coordinates._

A final question is what string will I use for the 'addArchRGenome' call? 'addArchRGenome("GRCh38")'?

**Comment by rcorces** [comment, marked as answer]
We've adopted TxDb and OrgDb as a standard to make our lives easier. This means we off-load any questions about those packages to the broader community. You should be able to find instructions on how to create your own TxDb object from a GTF file online if that is what you would like to do. You need to make very sure that your genome and gene annotation match correctly with the genome used for alignment.

When using a custom genome, you do not use `addArchRGenome()`. You pass your genome and gene annotations to the arguments of `createArrowFiles()` and `ArchRProject()`

https://www.archrproject.com/reference/createArrowFiles.html
https://www.archrproject.com/reference/ArchRProject.html
